searchcore/src/vespa/searchcore/proton/server/lid_space_compaction_job_take2.h

```diff
@@ -22,19 +30,38 @@ namespace proton::lidspace {
  * Moves documents from higher lids to lower lids. It uses a BucketExecutor that ensures that the bucket
  * is locked for changes while the document is moved.
  */
-class CompactionJob : public LidSpaceCompactionJobBase, public std::enable_shared_from_this<CompactionJob>
+class CompactionJob : public BlockableMaintenanceJob,
```

Comment:
As a follow-up, consider renaming the files to lid_space_compaction_job.h and lid_space_compaction_job.cpp

Reply:
Yes, will do.